ephios/core/notifications/types.py

```diff
@@ -0,0 +1,340 @@
+            slug=cls.slug,
+            user=user,
+            data=dict(participation_id=participation.id, email=participation.participant.email),
+        )
+
+    @classmethod
+    def get_subject(cls, notification):
+        return _("Participation confirmed")
+
+    @classmethod
+    def as_plaintext(cls, notification):
+        shift = AbstractParticipation.objects.get(
+            id=notification.data.get("participation_id")
+        ).shift
+        return _("Your participation for {shift} is now confirmed.").format(shift=shift)
+
+
+class ParticipationRejectedNotification(AbstractNotificationHandler):
+    slug = "ephios_participation_rejected"
+    title = _("Your participation has been rejected")
+
+    @classmethod
+    def send(cls, participation: AbstractParticipation):
+        user = (
+            participation.user
+            if participation.get_real_instance_class() == LocalParticipation
+            else None
+        )
+        Notification.objects.create(
+            slug=cls.slug,
+            user=user,
+            data=dict(participation_id=participation.id, email=participation.participant.email),
+        )
+
+    @classmethod
+    def get_subject(cls, notification):
+        return _("Participation rejected")
+
+    @classmethod
+    def as_plaintext(cls, notification):
+        shift = AbstractParticipation.objects.get(
+            id=notification.data.get("participation_id")
+        ).shift
+        return _("Your participation for {shift} has been rejected by a responsible user.").format(
+            shift=shift
+        )
+
+
+class ResponsibleParticipationRequested(AbstractNotificationHandler):
+    slug = "ephios_participation_responsible_requested"
+    title = _("A participation has been requested for your event")
+
+    @classmethod
+    def send(cls, participation: AbstractParticipation):
+        responsible_users = get_users_with_perms(
+            participation.shift.event, only_with_perms_in=["change_event"]
+        ).distinct()
+        disposition_url = urljoin(
+            settings.SITE_URL,
+            reverse("core:shift_disposition", kwargs=dict(pk=participation.shift.pk)),
+        )
+        notifications = []
+        for user in responsible_users:
+            notifications.append(
+                Notification(
+                    slug=cls.slug,
+                    user=user,
+                    data=dict(
+                        participation_id=participation.id,
+                        disposition_url=disposition_url,
+                    ),
+                )
+            )
+        Notification.objects.bulk_create(notifications)
+
+    @classmethod
+    def get_subject(cls, notification):
+        participation = AbstractParticipation.objects.get(
+            id=notification.data.get("participation_id")
+        )
+        participation_state = participation.get_state_display()
+        return _("Participation {state}").format(state=participation_state)
```

Comment:
Again, I think I prefer more context in the subject, like participant name and event title. 
Using the state display here is a nice idea, lets hope it works in translation.